src/convertToObject.js

```diff
@@ -12,7 +12,29 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const splitedString = sourceString.split(';');
+
+  const splitedProperties = splitedString.map(words =>
+    words.split(':')
+  );
+
+  const filtredProperties = splitedProperties.filter(value =>
+    value.length === 2
+  );
+
+  const trimmedProperties = filtredProperties.map(
+    ([key, value]) => [key.trim(), value.trim()]
+  );
+
+  for (const property of trimmedProperties) {
+    if (property.length > 1) {
+      result[property[0]] = property[1];
+    }
```

Comment:
```suggestion
      result[property[0]] = property[1];
```
Уже отфильтровали все непустые свойста.